src/app/project/project-documents/project-documents.component.ts

```diff
@@ -795,6 +795,13 @@ export class ProjectDocumentsComponent implements OnInit, OnDestroy {
       this.filterForUI[filter] = this.filterForUI[filter].filter(option => option._id !== item._id);
     }
 
+    public filterCompareWith(filter: any, filterToCompare: any)
```

Comment:
since we're using this method in three places, can we put it into the shared utils?

Reply:
So, turns out there's a state issue when the compareWith is executed. It's triggering after resolve, but before the Utils injection, so Utils will be null when it executes. I'm moving the comparator back into the individual classes for the time being until I can find a sensible work around outside of instantiating Utils in a function